### ISSUE TITLE
Skip To Content Unsuccessful in Mobile View

### ISSUE THREAD

**Issue by foxannefoxanne** [cerner/terra-framework#735]
# Bug Report

## Description

- While testing with screen readers in Mobile Browsers (Voiceover in iOS/Safari and Talkback in Android/Chrome with Pixel), the hidden 'skip to content' appears unsuccessful. 
- In iOS it will either open up the navigation or click on the logo (which in our app acts as a 'home link')
- In Android, it has no effects
- The skip to content button does not appear in mobile browsers. 
- Testing this morning on Safari with VoiceOver, Skip to Content was successful in the fullscreen view, but failed once the browser was shrunk down to a smaller size where the hamburger menu was present. 

## Steps to Reproduce
1. Navigate to https://engineering.cerner.com/terra-framework/#/ on mobile device 
2. Turn on Screen Reader Capabilities on mobile device
3. Navigate to where 'Skip to Main Content' is read using screen reader and select
4. Note that click does not seem to have any affect. 

## Expected Behavior
Skip to Content should direct users to main content in all browsers

## Environment
* Component Name and Version: terra-application-header-layout, 3.5.0 in our Component, also reproducible on https://engineering.cerner.com/terra-framework/#/
* Browser Name and Version:  Chrome 74 in Mobile, Safari 12 in iOS, Desktop
* Node/npm Version: lts/carbon in our app
* Webpack Version: 3.11.0
* Operating System and version: Reproducible on various android and ios devices, as well as Mojave on the Desktop

**Comment by bjankord** [comment]
Planning to investigate this and see what I can figure out on this issue.

**Comment by bjankord** [comment]
It looks like the common markup pattern for skip links may be broken on iOS and Android when using their respective screen readers - (VoiceOver and TalkBack). More info in this article: https://axesslab.com/skip-links/

* [A bug has been logged to Apple about in-page links not working as intended when using VoiceOver](https://bugs.webkit.org/show_bug.cgi?id=179011)
* [A bug has been logged to Google about broken skip links usage](https://bugs.chromium.org/p/chromium/issues/detail?id=657157)

**Comment by alex-bezek** [comment]
@bjankord thanks for the investigation!
Looking at this
![image](https://user-images.githubusercontent.com/8029578/60139598-1bce0d80-9774-11e9-9f66-714584d86599.png)

what is terra's standpoint on this? Do we wait for browser vendors to fix it, or implement one of the hacks they listed?

**Comment by bjankord** [comment]
The recommendation we've received from @scottwilmarth on items like this is to code the [common / recommended accessibility markup patterns](https://w3c.github.io/aria-practices/). If there is assistive technology that doesn't work with the common accessible markup patterns, we shouldn't try to customize the code to solve for a specific browser, screen reader, device bug/inconsistency.

The way we've implemented the skip to main content functionality is a fairly standard way based on resources I've seen for coding this type of functionality. 

Based on this, I'm planning to close this issue as a won't fix.  This stems from an issue on the browser side.  Sites created with the terra-application-layout should still be functional without a change.  Users can navigate to the main content landmark region due to the usage of `<main role="main" />` element used in the layout. As browsers correct the issues that have been logged about inconsistencies with the skip to content functionality, the terra-application-layout should continue to work as intended.
